### PR TITLE
Mntor-1630: Breaches GET/PUT

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-1630, MNTOR-1627

# Description
* Get breaches for user
* Modify breach resolution for user

Porting the following endpoints to NextJS
`GET http://localhost:6060/api/v1/user/breaches`
`PUT http://localhost:6060/api/v1/user/breaches`

# Screenshot (if applicable)

Not applicable.

# How to test

1. Before login, test redirect. If you go to `http://localhost:6060/api/v1/user/breaches` without logging in, you will be redirected to home
2. Login
3. Use the `GET` call to get the list of breaches for your account
4. Test `PUT` by sending the following body:
```json
{
"affectedEmail": "your@email.com",
"breachId": "your breach id",
"resolutionsChecked": ["...", "..."]
}
```

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/api/v1/user/breaches/route.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { NextRequest, NextResponse } from "next/server";
+import { getToken } from "next-auth/jwt";
+import { getBreaches } from "../../../../functions/server/getBreaches";
+import { getAllEmailsAndBreaches } from "../../../../../utils/breaches";
```
Comment (mansaj) [inline src/app/api/v1/user/breaches/route.ts:8]: We will be moving these `utils` into the nextjs folder

File: src/app/api/v1/user/breaches/route.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        currentBreaches[0].DataClasses.includes(val)
+      );
+      if (!isSubset) {
+        return NextResponse.json({
+          success: false,
+          message: `Error: the resolutionChecked param contains more than allowed data types: ${resolutionsChecked}`,
+        });
+      }
+
+      // /* new JsonB:
```
Comment (rhelmer) [inline src/app/api/v1/user/breaches/route.ts:105]: I know this is from the existing breaches controller, but is this comment still relevant for anything? I'm not exactly sure what it's referring to.

**Review by rhelmer — CHANGES_REQUESTED** [review]
It's really nice how we can port the existing controller code pretty much as-is! I tried the `GET` method (after authenticating so I had a valid JWT cookie from the `nextjs` branch) and it worked correctly.

File: src/app/api/v1/user/breaches/route.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  setBreachResolution,
+} from "../../../../../db/tables/subscribers";
+import appConstants from "../../../../../appConstants";
+
+// Get breaches data
+export async function GET(req: NextRequest) {
+  const token = await getToken({ req });
+  if (token) {
+    // Signed in
+    console.debug("JSON Web Token", JSON.stringify(token, null, 2));
```
[the remark this answers is not in the document]
Comment (mansaj) [reply src/app/api/v1/user/breaches/route.ts]: Fixed thanks

**Pull request merged**